### PR TITLE
fix CKV_AZURE_116 to reflect changes in v2.97.0 of Terraform Azure provider

### LINKED ISSUE #3006: CKV_AZURE_116 fails with Terraform Azure provider >= v.2.97.0
**Describe the issue**
This issue is regarding CKV_AZURE_116.
Since v2.97.0 of the Azure Terraform provider, the Azure policies add-on is enabled by setting

```terraform
azure_policy_enabled = true
```

as a top-level property in a `azurerm_kubernetes_cluster` resource definition, instead of the following

```terraform
addon_profile {
  azure_policy {
    enabled = true
  }
}
```
The check should be updated to reflect the change.

**Examples**
```terraform
resource "azurerm_kubernetes_cluster" "example" {
  azure_policy_enabled = true
}
```
should pass the check for CKV_AZURE_116.

**Version:**
2.0.1160

**Additional context**
None

I will submit a PR to fix this.

### PR DESCRIPTION
**Pull request by dommaes**
fixes #3006 

Unfortunately, I couldn't find the place to update the docs at https://docs.bridgecrew.io/docs/ensure-that-aks-uses-azure-policies-add-on. If someone can point me to the corresponding place, I will also update the docs.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by gruebel** [comment]
hi @dommaes thanks for the contribution 🚀 the docs are separate from the repository and you just need to click on **Suggest Edits**
regarding your changes, I'm totally up for only supporting the newer provider versions, but sadly our users need usually a bit longer to adapt to the new changes. I have to consult with my colleagues on how to best approach the transition time.
But thanks a lot for contributing and I will get back to you 🍻

**Comment by dommaes** [comment]
Thanks for the explanation, I totally get that.

In order to support both the old and the new version, what about reading the provider version from the Terraform lock file?
With that it would be possible to implement the check in a version dependent way.

I suppose this would turn out to be a bit more complex than it sounds.
Still, any thoughts on that?

**Comment by rotemavni** [comment]
Hi @dommaes, thank you for your contribution!
I have a suggestion on how to support both versions.

Currently the check extends `BaseResourceValueCheck` and my suggestion is to convert it to `BaseResourceCheck`. The check will check if one of the attributes exists and equals `True`. 
In case one of the attributes exist, set `self.evaluated_keys = [<found attribute>]`.

What do you think about this idea?

**Comment by dommaes** [comment]
I implemented your suggestion, thanks a lot for that @rotemavni! =)

File: checkov/terraform/checks/resource/azure/AKSUsesAzurePoliciesAddon.py [hunk 1]
```diff
@@ -1,17 +1,31 @@
...
+        azure_policy_enabled = conf.get("azure_policy_enabled", [None])[0]
+        if azure_policy_enabled:
+            self.evaluated_keys = ["azure_policy_enabled"]
+            return CheckResult.PASSED
+        # up to and including Azure provider v2.96.0
+        addon_profile = conf.get("addon_profile", [None])[0]
+        if addon_profile and isinstance(addon_profile, dict):
+            azure_policy = addon_profile.get("azure_policy", [None])[0]
+            if azure_policy and isinstance(azure_policy, dict) and azure_policy.get("enabled", [None])[0]:
+                self.evaluated_keys = ["addon_profile/[0]/azure_policy/[0]/enabled"]
```
Comment (gruebel) [inline checkov/terraform/checks/resource/azure/AKSUsesAzurePoliciesAddon.py:26]:
```suggestion
        self.evaluated_keys = ["addon_profile/[0]/azure_policy/[0]/enabled"]
        addon_profile = conf.get("addon_profile", [None])[0]
        if addon_profile and isinstance(addon_profile, dict):
            azure_policy = addon_profile.get("azure_policy", [None])[0]
            if azure_policy and isinstance(azure_policy, dict) and azure_policy.get("enabled", [None])[0]:
```
just move the `evaluated_keys` to the beginning of the condition check 🙂

**Review by gruebel — COMMENTED** [review]
other than that, great job 🚀

File: checkov/terraform/checks/resource/azure/AKSUsesAzurePoliciesAddon.py (hunk 1 above)
Comment (dommaes) [reply checkov/terraform/checks/resource/azure/AKSUsesAzurePoliciesAddon.py]: Done, thanks for the suggestion =)

**Review by gruebel — APPROVED** [review]
thanks, great work 🥇

**Review by rotemavni — APPROVED** [review]

**Pull request merged**